src/commands/deploy.ts

```diff
@@ -3,28 +3,87 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-// tslint:disable-next-line:no-require-imports
-import WebSiteManagementClient = require('azure-arm-website');
+import * as fse from 'fs-extra';
+import * as path from 'path';
 import * as vscode from 'vscode';
+import { SiteWrapper } from 'vscode-azureappservice';
 import { AzureFunctionsExplorer } from '../AzureFunctionsExplorer';
-import { IUserInterface } from '../IUserInterface';
+import { NoPackagedJavaFunctionError } from '../errors';
+import { IUserInterface, Pick } from '../IUserInterface';
 import { localize } from '../localize';
 import { FunctionAppNode } from '../nodes/FunctionAppNode';
 import { getWebSiteClient } from '../nodes/SubscriptionNode';
+import { TemplateLanguage } from '../templates/Template';
+import { cpUtils } from '../utils/cpUtils';
+import { getProjectType } from '../utils/project';
 import * as workspaceUtil from '../utils/workspace';
 import { VSCodeUI } from '../VSCodeUI';
 
+/* tslint:disable:no-require-imports */
+import WebSiteManagementClient = require('azure-arm-website');
+import StringDictionary = require('azure-arm-website/lib/models/StringDictionary');
+
 export async function deploy(explorer: AzureFunctionsExplorer, outputChannel: vscode.OutputChannel, context?: FunctionAppNode | vscode.Uri, ui: IUserInterface = new VSCodeUI()): Promise<void> {
     const uri: vscode.Uri | undefined = context && context instanceof vscode.Uri ? context : undefined;
     let node: FunctionAppNode | undefined = context && context instanceof FunctionAppNode ? context : undefined;
 
-    const folderPath: string = uri ? uri.fsPath : await workspaceUtil.selectWorkspaceFolder(ui, localize('azFunc.selectZipDeployFolder', 'Select the folder to zip and deploy'));
+    let folderPath: string = uri ? uri.fsPath : await workspaceUtil.selectWorkspaceFolder(ui, localize('azFunc.selectZipDeployFolder', 'Select the folder to zip and deploy'));
 
     if (!node) {
         node = <FunctionAppNode>(await explorer.showNodePicker(FunctionAppNode.contextValue));
     }
 
     const client: WebSiteManagementClient = getWebSiteClient(node);
+    const siteWrapper: SiteWrapper = node.siteWrapper;
+    const languageType: string = await getProjectType(folderPath);
+    if (languageType === TemplateLanguage.Java) {
+        folderPath = await getJavaFolderPath(outputChannel, folderPath, ui);
```

Comment:
So if I understand this correctly, I can _only_ have Java functions in my project? Seems like other languages would not get deployed

Reply:
Emmm, actually, I think yes you can (at least partially). For example, you can have a js function named blobTriggerJs. That means you will have a folder named blobTriggerJs where contains the js source and function property files. If you select this blobTriggerJs folder, this folder will be zipped and deployed (because the blobTriggerJs folder has no pom file in it).

But if we want to deploy all the functions in the project folder in one deployment (choose the project root folder), no we cannot achieve that.

Reply:
You can't deploy blobTriggerJs by itself - you have to deploy a project (that contains functions).

That's fine - let's just make sure to only display Java function templates if the current project is Java